### ISSUE TITLE
RDKit is a hard dependency -- README is wrong

### ISSUE THREAD

**Issue by pckroon** [deepchem/deepchem#1528]
Hello!
I tried to install DeepChem v2.1.0 from source (using pip) in a separate virtualenv. I know this is *not* the recommended way.
The README states that RDKit is a "soft" dependency, but it really isn't it seems.

```
mkvirtualenv deepchem
workon deepchem
git clone https://github.com/deepchem/deepchem/
cd deepchem
git checkout 2.1.0
pip install pandas joblib sklearn numpy tensorflow
pip install .
python
```
Next, in python:
```python
import deepchem
```
```
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import deepchem
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<PATH>/deepchem/deepchem/__init__.py", line 9, in <module>
    import deepchem.data
  File "<PATH>/deepchem/deepchem/data/__init__.py", line 9, in <module>
    from deepchem.data.datasets import pad_features
  File "<PATH>/deepchem/deepchem/data/datasets.py", line 12, in <module>
    from deepchem.utils.save import save_to_disk, save_metadata
  File "<PATH>/deepchem/deepchem/utils/__init__.py", line 17, in <module>
    from rdkit import Chem
ModuleNotFoundError: No module named 'rdkit'
>>> 
```

And unfortunately installing rdkit using pip is a *pain*.
The fastest fix for this issue would be to change the README to state that rdkit is a hard dependency, rather than a soft one. The painful fix would be to guard all the imports and mock the API.

**Comment by peastman** [comment]
It used to be a hard dependency.  Since 2.1 was released, we changed it to be a soft dependency.  If you use the latest code from github, it should work as you expect.  Also, we're getting ready to release 2.2 (hopefully this week).

**Comment by pckroon** [comment]
Thanks for the rapid response (and with the best possible answer!)
Unfortunately, it doesn't seem to work :(

```bash
> git checkout master
> pip install -U .
> python
```
```python3
import deepchem
```
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<PATH>/deepchem/deepchem/__init__.py", line 9, in <module>
    import deepchem.data
  File "<PATH>/deepchem/deepchem/data/__init__.py", line 18, in <module>
    from deepchem.data.data_loader import DataLoader
  File "<PATH>/deepchem/deepchem/data/data_loader.py", line 19, in <module>
    from deepchem.feat import UserDefinedFeaturizer
  File "<PATH>/deepchem/deepchem/feat/__init__.py", line 15, in <module>
    from deepchem.feat.graph_features import ConvMolFeaturizer
  File "<PATH>/deepchem/deepchem/feat/graph_features.py", line 5, in <module>
    from rdkit import Chem
ModuleNotFoundError: No module named 'rdkit'
```

**Comment by nd-02110114** [comment]
I think this is fixed! we can close this issue.

**Comment by rbharath** [comment]
Yep, this is now fixed in HEAD by #1874! Will close this issue
